optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/SingleBenchmarkResult.java

```diff
@@ -60,6 +61,7 @@
     private Score score = null;
     private long timeMillisSpent = -1L;
     private long calculateCount = -1L;
+    private EnvironmentMode environmentMode = null;
```

Comment:
See this line in SolverConfig:
  EnvironmentMode environmentMode_ = defaultIfNull(environmentMode, EnvironmentMode.REPRODUCIBLE);

I would recommend to extract a method called determineEnvironmentMode() of that line and reuse that from the benchmark report.

In any case, every singleBResult for the same solverBResult will always have the same environmentMode, so solverBResult, right? Even in a aggregation this still holds true. This means that the singleBResult should not have a field environmentMode (due to rule of normalization).